src/uct/cuda/cuda_ipc/cuda_ipc_md.h

```diff
@@ -121,6 +122,7 @@ typedef struct {
     uct_cuda_ipc_md_handle_t  ph;     /* Memory handle of GPU memory */
     CUdeviceptr               d_bptr; /* Allocation base address */
     size_t                    b_len;  /* Allocation size */
+    unsigned long long        buffer_id; /* Buffer ID for the allocation */
```

Comment:
This is a duplicate field. `uct_cuda_ipc_lkey_t::ph` contains this information when `HAVE_CUDA_FABRIC` is true. I think it is better to change `uct_cuda_ipc_md_handle_t` into a structure if `HAVE_CUDA_FABRIC` is false. Then `uct_cuda_ipc_md_handle_t` will have `buffer_id` field regardless of `HAVE_CUDA_FABRIC`. And you can also get rid of `#if HAVE_CUDA_FABRIC` in `uct_cuda_ipc_map_memhandle`.

Reply:
@rakhmets I've addressed this comment in https://github.com/openucx/ucx/pull/10405/commits/06107aa0585f868fc153e5a3bb3dd42728ac5b27. Can you please provide another review?